drv/sidecar-mainboard-controller/src/tofino2.rs

```diff
@@ -544,6 +564,47 @@ bitfield! {
     pub pcie_lanes, set_pcie_lanes: 7, 4;
 }
 
+#[derive(Copy, Clone, PartialEq, Eq, From, FromPrimitive, AsBytes)]
+#[repr(u32)]
+pub enum TofinoPcieResetOptions {
+    ControllerOnly = 0b00,
+    ControllerAndPhyLanes = 0b10,
+    EntirePhyAndController = 0b11,
+}
+
+impl From<TofinoPcieResetOptions> for u32 {
+    fn from(r: TofinoPcieResetOptions) -> Self {
+        r as u32
+    }
+}
+
+impl From<u32> for TofinoPcieResetOptions {
+    fn from(v: u32) -> Self {
+        match v {
+            0b10 => TofinoPcieResetOptions::ControllerAndPhyLanes,
+            0b11 => TofinoPcieResetOptions::EntirePhyAndController,
+            _ => TofinoPcieResetOptions::ControllerOnly, // Safe default.
+        }
```

Comment:
We can piggyback on the `FromPrimitive` derivation:
```suggestion
        Self::from_u32(v).unwrap_or(TofinoPcieResetOptions::ControllerOnly)
```

Reply:
Nice, thanks!